deepchecks/nlp/metric_utils/token_classification.py

```diff
@@ -0,0 +1,198 @@
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Utils module containing scorers for computing token classification metrics."""
+import typing as t
+from collections.abc import Sequence
+
+import numpy as np
+from seqeval.metrics import accuracy_score, f1_score, precision_score, recall_score
+from seqeval.scheme import Token
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.metrics import make_scorer
+
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp.text_data import TTokenLabel
+
+__all__ = ['get_default_token_scorers']
+
+
+DEFAULT_AVG_SCORER_NAMES = ('f1_macro', 'recall_macro', 'precision_macro')
+DEFAULT_PER_CLASS_SCORER_NAMES = ('f1_per_class', 'f1_per_class', 'f1_per_class')
+
+
+class SpanAligner:
+    """An object for transforming the labels and predictions of a specific dataset"""
+
+    def __init__(self, strict_alignment: bool = True, proba_threshold: float = 0.):
+        """Initialize the SpanAligner object.
+
+        Parameters:
+        -----------
+        strict_alignment: bool, default: True
+            If true, predictions are considered only if their spans match exactly any span in the labels. If false,
+            predictions are considered if they overlap any label span.
+        proba_threshold: float, default: 0
+            The minimal probability score for a prediction to be considered.
+        """
+        if not isinstance(strict_alignment, bool):
+            raise TypeError('strict_alignment must be a boolean')
+        if not strict_alignment:
+            # TODO: Implement non-strict alignment, probably using https://github.com/biocore-ntnu/ncls or similar
+            raise NotImplementedError('Non Strict alignment is not yet implemented')
+        if not isinstance(proba_threshold, float):
+            raise TypeError('proba_threshold must be a float')
+        if proba_threshold < 0 or proba_threshold > 1:
+            raise ValueError('proba_threshold must be between 0 and 1')
+
+        self.strict_alignment = strict_alignment
+        self.proba_threshold = proba_threshold
+        self.processed_labels = None
+        self.processed_predictions = None
+        self.classes = set()
+
+    def fit(self, y_true: TTokenLabel, y_pred: 'TTokenPred'):
+        """Fit the SpanAligner object to the labels and predictions of a specific dataset."""
+        self.processed_labels, self.processed_predictions = self._spans_to_token_list(y_true, y_pred)
+
+        # Post process class list to match the class order received by seqeval metrics
+        self.classes.discard('O')
+        self.classes = sorted(list(self.classes))
+
+    def fit_transform(self, y_true: TTokenLabel, y_pred: 'TTokenPred'
```

Comment:
Also add a transform method and have the fit_transform use it

Reply:
Implemented differently (called it get_transformed so that it's clear it does not do the transformation, just return cached)